samples/todos/src/main/java/com/toggl/komposable/sample/todos/Todos.kt

```diff
@@ -0,0 +1,80 @@
+package com.toggl.komposable.sample.todos
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.toggl.komposable.architecture.ReduceResult
+import com.toggl.komposable.architecture.Reducer
+import com.toggl.komposable.extensions.withSuspendEffect
+import com.toggl.komposable.extensions.withoutEffect
+import kotlinx.coroutines.delay
+import java.util.UUID
+
+sealed class TodosAction {
+    data object AddTodoButtonTapped : TodosAction()
+    data class FilterChanged(val filter: Filter) : TodosAction()
+    data object ClearCompletedButtonTapped : TodosAction()
+    data object SortCompletedTodos : TodosAction()
+    data class Delete(val ids: Set<UUID>) : TodosAction()
+    data class Todo(val index: Int, val action: TodoAction) : TodosAction()
+}
+
+enum class Filter {
+    All,
+    Active,
+    Completed,
+}
+data class TodosState(
+    val todos: List<TodoState> = emptyList(),
+    val filter: Filter = Filter.All,
+) {
+    val filteredTodos: List<TodoState> = when (filter) {
+        Filter.All -> todos
+        Filter.Active -> todos.filter { !it.isComplete }
+        Filter.Completed -> todos.filter { it.isComplete }
+    }
+}
+
+class TodosReducer : Reducer<TodosState, TodosAction> {
+    override fun reduce(state: TodosState, action: TodosAction): ReduceResult<TodosState, TodosAction> =
```

Comment:
We could add some "fake download" to each item to simulate the cancellable events and test that we can cancel a bunch of ongoing effects in one go.

Reply:
We should do this in tests, I want to keep this sample as close to https://github.com/pointfreeco/swift-composable-architecture/tree/main/Examples/Todos/Todos as possible.